packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -206,23 +206,24 @@ export class OdspDocumentService implements IDocumentService {
                 ? Promise.resolve(null)
                 : this.getWebsocketToken!(options);
             const joinSessionPromise = this.joinSession(requestWebsocketTokenFromJoinSession).catch((e) => {
-                let code: string | undefined;
-                try {
-                    code = e?.response ? JSON.parse(e?.response)?.error?.code : undefined;
-                } catch (error) {
-                    throw e;
-                }
-                switch (code) {
-                    case "sessionForbiddenOnPreservedFiles":
-                    case "sessionForbiddenOnModerationEnabledLibrary":
-                    case "sessionForbiddenOnRequireCheckout":
-                        // This document can only be opened in storage-only mode. DeltaManager will recognize this error
-                        // and load without a delta stream connection.
-                        this._policies = {...this._policies,storageOnly: true};
-                        throw new DeltaStreamConnectionForbiddenError(code);
-                    default:
-                        throw e;
+                const likelyFacetCodes = e as IFacetCodes;
+                if (likelyFacetCodes.facetCodes !== undefined) {
```

Comment:
For safety, consider replacing it with `if (Array.isArray(likelyFacetCodes.facetCodes))` check